Frameworks/Foundation/NSOperation.mm

```diff
@@ -119,10 +125,10 @@ - (BOOL)isReady {
  @Status Interoperable
 */
 - (void)addDependency:(id)operation {
-    std::lock_guard<std::recursive_mutex> lock(_dependenciesLock);
+    std::lock_guard<std::recursive_mutex> dependenciesLock(_dependenciesLock);
     [self willChangeValueForKey:@"dependencies"];
     [_dependencies addObject:operation];
```

Comment:
>_dependencies [](start = 5, length = 13)

do you need to do 

if ([operation isFinished]) {
            _readyCount++;
        }